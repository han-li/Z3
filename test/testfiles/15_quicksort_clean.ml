let rec qsort = function
  | [] -> []
  | pivot :: rest ->
      let is_less x = x < pivot in
      let left, right = List.partition is_less rest in
      qsort left @ [pivot] @ qsort right;;


let rec _print_int_list = function
    | [] -> ()
    | fst :: [] ->
        print_int fst; ();
    | fst :: rest ->
        print_int fst;
        print_string ",";
        _print_int_list rest;
        ()
;;

let print_int_list l = 
    print_string "[";
    _print_int_list l;
    print_string "]";
    print_newline ();
    ()
;;

print_int_list (qsort
[1;56;12;8;23;123;9423;9;39;4;5;12;53;43;94;239;12;30]);;
