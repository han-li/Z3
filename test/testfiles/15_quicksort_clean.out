[1,4,5,8,9,12,12,12,23,30,39,43,53,56,94,123,239,9423]
